File: arrays_and_strings/running_sum_of_1d_array.cpp
#include <gtest/gtest.h>

#include <algorithm>
#include <vector>

static std::vector<int> runningSumFA(std::vector<int> nums)
{
    std::vector<int> prefix {};
    prefix.reserve(nums.size());
    prefix.push_back(nums[0]);

    for (int i = 1; i < static_cast<int>(nums.size()); ++i)
    {
        prefix.push_back(nums[i] + prefix.back());
    }

    return prefix;
}

TEST(RunningSumTest, SampleTest)
{
    const std::vector<int> expected_output {1, 3, 6, 10};

    const auto resultFA = runningSumFA({1, 2, 3, 4});

    EXPECT_TRUE(std::equal(expected_output.cbegin(),
                           expected_output.cend(),
                           resultFA.cbegin()));
}

File: arrays_and_strings/running_sum_of_1d_array.java
public int[] runningSum1(int[] nums)
{
    int[] prefix = new int[nums.length];
    prefix[0] = nums[0];

    for (int i = 1; i < nums.length; i++)
    {
        prefix[i] = nums[i] + prefix[i - 1];
    }

    return prefix;
}

public int[] runningSum2(int[] nums)
{
    for (int i = 1; i < nums.length; i++)
    {
        nums[i] += nums[i - 1];
    }
    return nums;
}